### DISCUSSION TITLE
complex 3d geometry

### DISCUSSION THREAD

**Discussion by jav-ed** [lululxvi/deepxde#1131] [Q&A] [answered]
Hello everyone,
I would like to model a 3d aircraft, thus my question, is it possible to model a complex geometry in 3d. In [docs 3d geom](https://deepxde.readthedocs.io/en/latest/modules/deepxde.geometry.html#module-deepxde.geometry.geometry_3d) some 3d geometries can be chosen from. However, a 3d polygon function like: [ deepxde.geometry.geometry_2d.Polygon(vertices)](https://deepxde.readthedocs.io/en/latest/modules/deepxde.geometry.html#module-deepxde.geometry.geometry_2d) is required. Is there a 3d polygon function for generating complex 3d geometries?

Also, is there any other method I should know about to incorporate 3d geometries. The aircraft model should be available in [gmsh](https://gmsh.info/)

Thank you for your time and effort

**Comment by jav-ed** [comment, marked as answer]
I found [issue 161](https://github.com/lululxvi/deepxde/issues/161), which is similar to my question. It seems that there is no 3d polygon function.

**Comment by lululxvi** [comment]
Try `PointCloud` https://deepxde.readthedocs.io/en/latest/modules/deepxde.geometry.html#module-deepxde.geometry.pointcloud
